File: python/word-count/word_count.py
def count_words(sentence):
    sentence = sentence.replace('\n', ' ').replace('\t', ' ')
    sentence = ''.join([c if c.isalnum() or c == "'"  else ' ' for c in sentence])
    words = sentence.lower().strip("'").split()
    word = [e.strip("'") for e in words]
    wordcount = {}
    for i in word:
        if i in wordcount:
           wordcount[i] += 1
        else:
           wordcount[i] = 1
    return wordcount

File: julia/word-count/word-count.jl
using Printf

function wordcount(sentence)

	#Let's replace non-ascii characters
	sentence=replace(sentence, r"[\.,!\^\&\*\(\)~@#\\\$%\{\}\[\]:;\/\<\>]"=>" ")
  	sentence=replace(sentence, r"^'|[(?<=\s)']['(?=\s)]|'$"=>" ")

	#Let's replace Uppercase by lowercases
	sentence = lowercase(sentence)

	#Let's split on all non-letter/non-digit except appostrophe if it is in the middle of a word
	words = split(sentence, c -> !(isletter(c) || isdigit(c) || c == '''); keepempty = false)

	#Let's create a Dictionary out of the words in the sentence
	d = Dict{String, Int64}()
	for w in words
        	d[w] = count(j == w for j in words)
	end
    	d
end
